2_Scope_Closures/practice_new/closure_part3_holee.js

```diff
@@ -0,0 +1,140 @@
+"use strict"
+
+function formatTotal(display) {
+    if (Number.isFinite(display)) {
+        // constrain display to max 11 chars
+        let maxDigits = 11;
+        // reserve space for "e+" notation? 
+        if (Math.abs(display) > 99999999999) {
+            maxDigits -= 6; 
+        }
+        // reserve space for "-"?
+        if (display < 0) { 
+             maxDigits--;
+        }
+
+        // whole number?
+        if (Number.isInteger(display)) { 
+            display = display
+                .toPrecision(maxDigits) 
+                .replace(/\.0+$/,"");
+        }
+        // decimal
+        else {
+            // reserve space for "." 
+            maxDigits--;
+            // reserve space for leading "0"? 
+            if (
+                Math.abs(display) >= 0 &&
+                Math.abs(display) < 1 
+            ){
+                maxDigits--; 
+            }
+            display = display 
+                .toPrecision(maxDigits) 
+                .replace(/0+$/,"");
+            } 
+        }
+        else {
+            display = "ERR";
+        }
+    return display; 
+}
+
+function calculator() { 
+    var nbr = "";
+    var total = 0;
+    var is_equal = 0;
+    var operator = '';
+    
+    return function calc(key) {
+        if (is_equal) {
+            is_equal = 0;
+            if (is_number(key)) {
+                total = 0;
+                operator = '';
+            }
+        }
+        if (is_number(key)) {
+            nbr += key;
+            return (key);
+        }
+        if (is_operator(key)) {
+            if (operator === '') {
+                total = Number(nbr);
+                nbr = "";
+                return (operator = key);
+            }
+            total = calc_operator(total, operator, Number(nbr));
+            nbr = "";
+            return (operator = key);
+        }
+        if (key === '=') {
+            if (operator === '') {
+                total = Number(nbr);
+            } else {
+                total = calc_operator(total, operator, Number(nbr));
+                is_equal = 1;
+                nbr = "";
+            }
+            return (formatTotal(total));
+        }
+        return "";
+    }
+    
+    function is_number(str) {
+       return (str >= '0' && str <= '9');
+    }
+    
+    function is_operator(str) {
+        if (str === '+' || str === '-'
+        || str === '/' || str === '%'
+        || str === '*')
+            return (true);
+        return (false);
```

Comment:
```suggestion
    function is_operator(str) {
       return ['+', '-', '/', '%', '*'].includes(str)
     }
```
이런 식으로 코드를 작성하면 가독성을 해치지 않으면서 더 보기 좋은 코드를 작성할 수 있을거 같습니다!

Reply:
gim님 코드보면서 includes 존재를 알았습니다. 확실히 깔끔해 보이네요 좋은 코드 알려주셔서 감사합니다!